pandas/io/sql.py

```diff
@@ -1836,7 +1836,11 @@ def __init__(self, *args, **kwargs) -> None:
 
         # this will transform time(12,34,56,789) into '12:34:56.000789'
         # (this is what sqlalchemy does)
-        sqlite3.register_adapter(time, lambda _: _.strftime("%H:%M:%S.%f"))
+        def _adapt_time(t):
+            # This is faster than strftime
+            return f"{t.hour:02d}:{t.minute:02d}:{t.second:02d}.{t.microsecond:06d}"
```

Comment:
are there actual benchmarks for this?

Reply:
As explained above 

> I did not write an ASV for the SQL perf improvements, as the one in io/sql.py could make the job but the time type is actually not present, and all types are passed at once to the to_sql method. So if I add a column the perf will degrade. Another option is to split this ASV test into several by parametrizing the column to send.

I'll try to improve the one in io/sql.py to split it by column type. You'll let me know if this is ok.

Reply:
Done. See comment in the main thread.